### PR TITLE
Add @threefold/UI package

### PR DESCRIPTION
**Pull request by zaelgohary**
### Description

- Add 3script Editor
- Add PDF signer

### Related Issues

- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/1056
- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/1057

### Checklist

- [ ] Tests included
- [ ] Build pass
- [ ] Documentation
- [ ] Code format and docstrings
- [ ] Screenshots/Video attached (needed for UI changes)

### CODE REVIEW

**Review by AhmedHanafy725 — CHANGES_REQUESTED** [review]

File: packages/UI/src/components/ScriptEditor.vue [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+      const newLines = newText.split("\n");
+      lines.value = newLines;
+    };
+
+    const submitScript = async () => {
+      try {
+        const account = await ThreefoldWalletConnectorApi.selectDecryptedAccount();
+        const data = await sign(text.value, account?.mnemonic ?? "", KeypairType.sr25519);
+        const response = axios.post(props.dest, data);
+        console.log(response);
```
[the remark this answers is not in the document]
Comment (Mahmoud-Emad) [reply packages/UI/src/components/ScriptEditor.vue:84]: we set this for now to see the response

File: packages/UI/src/docs/pdf_viewer.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+<!-- You can install the PDF Signer Web Component using npm or include it via a script tag. -->
+
+<!-- ### npm
+
+To install via npm, use the following command:
+
+```bash
+npm install @threefold/UI
+``` -->
```
[the remark this answers is not in the document]
Comment (Mahmoud-Emad) [reply packages/UI/src/docs/pdf_viewer.md:29]: sure

File: packages/UI/src/docs/pdf_viewer.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+</body>
+</html>
+```
+
+Script Tag:
+
+1. Clone the `tfgrid-sdk-ts` repository.
+2. Navigate to the `repository/packages/UI` and run `yarn build-only`.
+3. A `dist` folder will be created. Copy the `threefold-ui.umd.js` file and include it in your scripts.
+4. Copy the `tailwind.css` to your side project.
```
[the remark this answers is not in the document]
Comment (Mahmoud-Emad) [reply packages/UI/src/docs/pdf_viewer.md:73]: inside the dist folder, gonna update

File: .github/workflows/threefpld_ui_build.yaml [hunk 4]
```diff
@@ -0,0 +1,45 @@
+# This workflow will do a clean install of node dependencies, cache/restore them, build the source code using yarn build-only
+
+name: Stats Build
```
Comment (AhmedHanafy725) [inline .github/workflows/threefpld_ui_build.yaml:3]: stats??

File: packages/UI/docs/pdf_viewer.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+<!-- ### Importing the PDF Signer
+
+Import the PDF Signer Web Component into your JavaScript file.
+
+```js
+import pdfSigner from "@threefold/UI";
+``` -->
+
+### Initializing the PDF Signer
```
Comment (AhmedHanafy725) [inline packages/UI/docs/pdf_viewer.md:29]: this the build instruction, not initialization

File: packages/UI/src/components/PDFSignerViewComponent.vue [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+      if (pdfData.value) {
+        if (!hasAccess.value) {
+          hasAccess.value = await ThreefoldWalletConnectorApi.requestAccess();
+        }
+        const account = await ThreefoldWalletConnectorApi.selectDecryptedAccount();
+        const data = await sign(pdfData.value, account?.mnemonic ?? "", KeypairType.ed25519);
+        const requestBody: PDFPostData = {
+          pdfUrl: props.pdfurl,
+          pubkey: account?.address || "",
+          signature: data,
```
Comment (AhmedHanafy725) [inline packages/UI/src/components/PDFSignerViewComponent.vue:134]: please check the return data from the sign method. it contains the address and the signature

File: .github/workflows/threefpld_ui_build.yaml
Comment (AhmedHanafy725) [inline .github/workflows/threefpld_ui_build.yaml:1]: please update the [workflow docs](https://github.com/threefoldtech/tfgrid-sdk-ts/blob/development_312_UI/docs/workflows.md) with the added workflow

File: yarn.lock
[the remark this answers is not in the document]
Comment (Mahmoud-Emad) [reply yarn.lock:1]: done

File: packages/UI/README.md
[the remark this answers is not in the document]
Comment (Mahmoud-Emad) [reply packages/UI/README.md:1]: Please point me with the actual usage of the `threefold UI` package to update it

File: packages/UI/src/components/PDFSignerViewComponent.vue [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+    const loadingPdf = ref(false);
+    const isError = ref(false);
+    const errorMessage = ref("");
+    const pdfData = ref<string>("");
+    const isAcceptDisabled = ref(true);
+    const loadingAcceptBtn = ref(false);
+    const isInstalled = ref();
+    const hasAccess = ref();
+
+    console.log("From setUp: ", props.pdfurl);
```
[the remark this answers is not in the document]
Comment (Mahmoud-Emad) [reply packages/UI/src/components/PDFSignerViewComponent.vue]: Should be removed

File: packages/UI/docs/pdf_viewer.md (hunk 5 above)
Comment (zaelgohary) [reply packages/UI/docs/pdf_viewer.md]: changed

File: .github/workflows/threefpld_ui_build.yaml
Comment (zaelgohary) [reply .github/workflows/threefpld_ui_build.yaml:1]: done.

File: .github/workflows/threefpld_ui_build.yaml (hunk 4 above)
Comment (zaelgohary) [reply .github/workflows/threefpld_ui_build.yaml]: edited.

File: packages/UI/docs/server_verification.md [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+}
+```
+
+#### Unverified Signature
+
+If the signature cannot be verified, the server will respond with an HTTP status code of 200 and a JSON response like this:
+
+```json
+{
+  "status": 200,
```
Comment (AhmedHanafy725) [inline packages/UI/docs/server_verification.md:57]: why do we return 200 in case of failure?

File: packages/UI/src/utils/sign.ts [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+
+  await waitReady();
+
+  const key = keyr.addFromMnemonic(mnemonic);
+  const publicKey = key.address;
+
+  const signed = key.sign(messageBytes);
+  const signature = Buffer.from(signed).toString("hex");
+
+  return { signature, publicKey };
```
Comment (AhmedHanafy725) [inline packages/UI/src/utils/sign.ts:29]: please return the signature only here

File: packages/UI/src/utils/types.ts [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+      case "main":
+        return NetworkEnv.main;
+      case "test":
+        return NetworkEnv.test;
+      case "qa":
+        return NetworkEnv.qa;
+      case "dev":
+        return NetworkEnv.dev;
+      default:
+        return NetworkEnv.dev;
```
Comment (AhmedHanafy725) [inline packages/UI/src/utils/types.ts:61]: the default should be mainnet

**Review by AhmedHanafy725 — COMMENTED** [review]
The example should be outside the `src` dir.

File: packages/UI/src/utils/sign.ts (hunk 9 above)
Comment (Mahmoud-Emad) [reply packages/UI/src/utils/sign.ts:29]: We need both of them since we have two provider

File: packages/UI/README.md [hunk 11]
```diff
@@ -0,0 +1,28 @@
+# UI
+
+Threefold UI is a project that contains reusable web components created by Threefold organization.
+
+## Components
+
+- [PDF Viewer](./docs/pdf_viewer.md)
+- [Script Editor](./docs/script_editor.md)
+- [Verification Server Exsample](./docs/server_verification.md)
+- [Web Component Exsample](./docs/web_component.md)
```
Comment (AhmedHanafy725) [inline packages/UI/README.md:10]: they are not components to be added here

Comment (AhmedHanafy725) [inline packages/UI/README.md:10]:
```suggestion
- [Verification Server Example](./docs/server_verification.md)
- [Web Component Example](./docs/web_component.md)
```

**Review by AhmedHanafy725 — APPROVED** [review]

**Pull request merged**